pkg/tree/iterator.go

```diff
@@ -0,0 +1,59 @@
+package tree
+
+import (
+	"github.com/ipld/go-ipld-prime"
+	basicnode "github.com/ipld/go-ipld-prime/node/basic"
+	"io"
+	"sync"
+)
+
+type pair struct {
+	key   []byte
+	value ipld.Node
+}
+
+func NewIterator() *Iterator {
+	return &Iterator{
+		result: make([]pair, 0),
+	}
+}
+
+type Iterator struct {
```

Comment:
Does golang have any standards for iterators we can work off of? What does the rest of IPLD use?

Would it make to use a channel here or expose the data as a channel?

Reply:
The iterator implements the MapIterator( https://github.com/ipld/go-ipld-prime/blob/1a409dc236dd1697fa787ca7166e43c884a57d0e/datamodel/node.go#L259 )interfaces in ipld node(map kind), so I reuse it in the search api and adl. Should we have more methods to support the iterator?